### ISSUE TITLE
Render comprehensive boundary labels

### ISSUE THREAD

**Issue by ZeLonewolf** [openmaptiles/openmaptiles#1216]
It is typical in paper maps to render names along the sides of boundary lines.  In the case where multiple boundaries converge (for example, country, state, city/town), it is typical to render multiple names along the line.  The current behavior of the `boundary` layer renders left/right labels with only the 3-letter country code.

The desired behavior is:
* Separate features are available for rendering at each `admin_level`
* A complete `name:XX` hierarchy is built for each boundary feature, so that they can be localized.

Below is an example of this behavior on paper maps:

(TODO)

It is not possible to add this capability to the current `boundary` label, which is structured on a line segment basis for styled rendering, with a single object per segment.  Instead, right-hand-rule normalized polygons are needed for boundary label styling.  Therefore, a new layer is required to accomplish this.

**Comment by nyurik** [comment]
The reason we collapse all boundary lines into one is because of non-solid boundary lines. For example, disputed boundaries are drawn as dashes/dots, but if there is more than one boundary (disputed country + region), the lower-priority "region" boundary overlaps the dashed "country" line, making it appear as a solid.

**Comment by ZeLonewolf** [comment]
I agree this is correct behavior, for line rendering of border lines. However, this approach is not sufficient for proper name labeling. That's why I'm proposing that an additional layer is needed for the labels themselves, which need a one to many relationship over a segment.

**Comment by nyurik** [comment]
thx @ZeLonewolf , makes sense.  I think we had a similar issue with multiple highway tags for a single geometry, but i'm not sure how it was solved.  Although highway tags don't need to be localized i think, so this might be different.
